src/main/scala/beam/agentsim/agents/vehicles/VehicleEnergy.scala

```diff
@@ -210,33 +223,35 @@ class VehicleEnergy(
     powerTrainPriority: PowerTrainPriority
   ): Double = {
     /*(Double, Option[Double]) = {*/
-    val BeamVehicle.FuelConsumptionData(
-      linkId,
-      vehicleType,
-      numberOfLanesOption,
-      _,
-      _,
-      _,
-      speedInMetersPerSecondOption,
-      _,
-      _,
-      _
-    ) = fuelConsumptionData
-    val numberOfLanes: Int = numberOfLanesOption.getOrElse(0)
-    val speedInMilesPerHour: Double = speedInMetersPerSecondOption
-      .map(convertFromMetersPerSecondToMilesPerHour)
-      .getOrElse(0)
-    val gradePercent: Double = linkIdToGradePercentMap.getOrElse(linkId, 0)
-    (powerTrainPriority match {
-      case Primary   => consumptionRateFilterStore.getPrimaryConsumptionRateFilterFor(vehicleType)
-      case Secondary => consumptionRateFilterStore.getSecondaryConsumptionRateFilterFor(vehicleType)
-    }).flatMap(
-        consumptionRateFilterFuture =>
-          getRateUsing(consumptionRateFilterFuture, numberOfLanes, speedInMilesPerHour, gradePercent)
-      )
-      .getOrElse(fallBack(fuelConsumptionData))
-    /*.map(x=>(x,Option(gradePercent)))
+    if (!vehicleEnergyMappingExistsFor(fuelConsumptionData.vehicleType)) { fallBack(fuelConsumptionData) } else {
+      val BeamVehicle.FuelConsumptionData(
+        linkId,
+        vehicleType,
+        numberOfLanesOption,
+        _,
+        _,
+        _,
```

Comment:
I believe is a good idea pass the parameters names whenever parameters >= 4

Reply:
I disagree - the names extracted are the ones I believe should be at focus here.